lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -389,10 +389,22 @@ defmodule ExUnit.Runner do
     context
   end
 
+  defp short_hash(module, test_name) do
+    (module <> "/" <> test_name)
+    |> :erlang.md5()
+    |> Base.encode16(case: :lower)
+    |> String.slice(0..7)
+  end
+
   defp create_tmp_dir!(test, extra_path, context) do
-    module = escape_path(inspect(test.module))
-    name = escape_path(to_string(test.name))
-    path = ["tmp", module, name, extra_path] |> Path.join() |> Path.expand()
+    module_string = inspect(test.module)
+    name_string = to_string(test.name)
+
+    module = escape_path(module_string)
+    name = escape_path(name_string)
+    short_hash = short_hash(module_string, name_string)
```

Comment:
If we're calculating a MD5 hash anyways, maybe better to save a couple of lines and just calculate the hash of `short_hash(module, name)`?

Reply:
that will defeat the purpose of the PR, as it will be calculating the same hash for "foo-bar" and "foo+bar" as they are already escaped.